### PR TITLE
potential Doc2Vec feature: reverse inference in doc2vec 

### PR DESCRIPTION
**Pull request by saraswatmks**
This feature is in accordance to this issue: [2459](https://github.com/RaRe-Technologies/gensim/issues/2459)

It does the following:
1. `predict_output_word` function take a document vector and returns a list of tuples having word and probability score.
2. The reference for this implementation has been taken from `Word2Vec.predict_output_word()` function.

### CODE REVIEW

**Comment by gojomo** [comment]
Thanks, looks promising! To distinguish from `predict_output_word()` (in both function, and to prevent overiding it), I'd suggest a slightly different name. Perhaps `predict_words()`? (Similarly, the two methods might be able to share some logic.)

Have you been able to "eyeball" some results on test datasets, to get an idea if (as conjectured in the feature-request) they work as a sort of `summary`? For example:

* do the top-n words for a known-document doc-vector seem related to the text? 
* if using a `Doc2Vec` model with trained word-vectors (like `dm=1` or `dm=0, dbow_words=1` mode), do these predicted-words, or the most-similar-words-to-the-doc-vector, better overlap with the actual words of the text?
* similarly, are the top-N most-predicted or the top-N most-similar closers via a measure like "word' mover's distance" to the actual text?

**Comment by saraswatmks** [comment]
@gojomo I appreciate your quick response. I haven't checked the sanity of results yet. I will do that now.

**Comment by saraswatmks** [comment]
@gojomo I've renamed the function to `predict_words`. Also, the common logic I moved to parent class `BaseWordEmbeddingsModel`. I did the "eyeball" testing of results. They look fine. The similar words come from the same document whose document vector is provided, so yes there is significant overlap. The words overlap varies with different parameters `dm=1 or dm=0, dbow_words=1`. 

**P.S** - The pipeline is failing because scikit-learn no longer supports build on python 2.7

File: gensim/models/base_any2vec.py [hunk 1]
```diff
@@ -1470,3 +1470,58 @@ def evaluate_word_pairs(self, pairs, delimiter='\t', restrict_vocab=300000,
 
         """
         return self.wv.evaluate_word_pairs(pairs, delimiter, restrict_vocab, case_insensitive, dummy4unknown)
+
+    def _find_similar_words(self, doc_vector=None, context_words_list=None, topn=None):
```
Comment (gojomo) [inline gensim/models/base_any2vec.py]: These aren't necessarily "similar" words (as in the various other "similarity" methods), but *predicted* words, and the method name should reflect that. Separately, it's smewhat weird factoring to have a `doc_vector` input for a method in a Base class, and also weird to have two alternate parameters that are (not usually/sanely?) used together – that's a hint this *isn't* the truly shared forward-prop functionality – which would start with the shared path the method below does with `use_vector`.

**Comment by gojomo** [comment]
As a speculative/unproven feature – even though I think it's a promising idea, or wouldn't have suggested it – I think there needs to be some published example(s) of its use/value to justify its integration. Ideally, this would be either a demo notebook (or extension of some existing notebook to show this functionality).

**Comment by saraswatmks** [comment]
@gojomo  I agree with you. I will make the requested changes along with a demo notebook.

**Comment by mpenkov** [comment]
@saraswatmks How's it going? Could you please give us an estimate of when you can finish this PR? Otherwise, we will have to close it due to lack of activity.

**Pull request closed without merging**